src/mumble/AudioInput.cpp

```diff
@@ -548,6 +548,30 @@ void AudioInput::initializeMixer() {
 }
 
 void AudioInput::addMic(const void *data, unsigned int nsamp) {
+	// check if there are trailing zeros in the sample and adjust nsamp
+	if (nsamp > 0) {
+		if (eMicFormat == SampleFloat) {
+			const auto *float_data = reinterpret_cast< const float * >(data);
+			unsigned int sample_end_pos = 0;
+			for (unsigned int i = 0; i < nsamp; ++i) {
+				if (float_data[i] != 0.0) {
+					sample_end_pos = i > sample_end_pos ? i : sample_end_pos;
```

Comment:
Maybe a bit more clearer (semantically):
```suggestion
					sample_end_pos = std::max(i, sample_end_pos);
```

I am wondering however if there even could be a situation in which `i < sample_end_pos` here. From a first glance, I think it should be save to simply assign `i` to `sample_end_pos` without this check :thinking:

Reply:
sample_end_pos is initialized to be zero, and according to 558, it can only be smaller or equal to sample_end_pos, I believe.

Reply:
Make sense.